taxcalc/decorators.py

```diff
@@ -283,7 +285,7 @@ def make_wrapper(func):
         try:
             jit_args = inspect.getargspec(jit).args + ['nopython']
         except TypeError:
-            print ("This should only be seen in RTD, if not install numba!")
+            #print ("This should only be seen in RTD, if not install numba!")
```

Comment:
On Mon, Jun 22, 2015 at 9:42 AM, T.J. Alumbaugh <notifications@github.com>
wrote:

> In taxcalc/decorators.py
> <https://github.com/OpenSourcePolicyCenter/Tax-Calculator/pull/285#discussion_r32934642>
> :
>
> > @@ -283,7 +285,7 @@ def make_wrapper(func):
> >          try:
> >              jit_args = inspect.getargspec(jit).args + ['nopython']
> >          except TypeError:
> > -            print ("This should only be seen in RTD, if not install numba!")
> > +            #print ("This should only be seen in RTD, if not install numba!")
>
> I would definitely advocate for something other than just commenting out
> the print statement. One way to handle this is to attempt the import of
> jit at the top and if numba is not found, produce a dummy function.
> Something like this:
>
> # Numba is not available when building on ReadTheDocs
> try:
>     from numba import jit
> except ImportError:
>     jit = lambda x: return x
>
> then this line inside the function would simply be:
>
>     jit_args = inspect.getargspec(jit).args + ['nopython']
>
> because jit is defined in either case. How does that sound?
>

Does the above suggested code revision mean that a user does not need numba
as part of the Python distribution to execute the Tax Calculator?  Because
with
your suggested code revision jit is just a dummy function that does nothing
when
numba in not installed?  Right?  Or am I missing something?

 If that is your intention, then I think it is an improvement in the design
of the Tax Calculator.  I have always thought that the way things should
work is that if numba is installed tax calculations will execute fast, but
if
not installed the calculator should run and get the same answers but
take longer to execute.  Is that what you are suggesting?  If so, then
I like your suggested revision.  If not, see my (actually Matt's) question
below.

What Matt asked you was something different: is there (or should there be)
at the beginning of the Tax Calculator logic code that checks for numba
installation and stops Tax Calculator execution with an error message when
numba is not part of the installed Python distribution.  I'm interested in
your
answer to that question.